fixture-monkey-api/src/main/java/com/navercorp/fixturemonkey/api/generator/OptionalContainerPropertyGenerator.java

```diff
@@ -48,21 +50,19 @@ public final class OptionalContainerPropertyGenerator implements ContainerProper
 	public ContainerProperty generate(ContainerPropertyGeneratorContext context) {
 		Property property = context.getProperty();
 
-		AnnotatedType valueAnnotatedType = getValueAnnotatedType(property);
-		Property valueProperty = new ElementProperty(
-			property,
-			valueAnnotatedType,
-			0,
-			0
-		);
+		AnnotatedType valueAnnotatedType = getOptionalValueAnnotatedType(property);
+		Type valueType = valueAnnotatedType.getType();
+
+		Property childProperty = new OptionalProperty(valueType, valueAnnotatedType);
+		SingleElementProperty singleElementProperty = new SingleElementProperty(childProperty, valueAnnotatedType);
```

Comment:
I think it could be enough for `SingleElementProperty`.
Is it not work if only one `SingleElementProperty` like below?

```java
AnnotatedType valueAnnotatedType = getOptionalValueAnnotatedType(property);
		Type valueType = valueAnnotatedType.getType();

SingleElementProperty singleElementProperty = new SingleElementProperty(valueAnnotatedType);
```

I do not know why `SingleElementProperty` needs `valueAnnotatedType' as it is the annotated type of `childProperty'.

I think I only need one or the other between `childProperty` and `valueAnnotatedType`.

I'm not sure it's right time to split into `OptionalProperty` and `SupplierProperty`.

Reply:
> I think it could be enough for SingleElementProperty.
> Is it not work if only one SingleElementProperty like below?
> 
> ```
> AnnotatedType valueAnnotatedType = getOptionalValueAnnotatedType(property);
> 		Type valueType = valueAnnotatedType.getType();
> 
> SingleElementProperty singleElementProperty = new SingleElementProperty(valueAnnotatedType);
> ```
> 
> I do not know why SingleElementProperty needs valueAnnotatedType' as it is the annotated type of childProperty'.


if SingleElementProperty(remove property field) like below, test broken and i think resson is cannot get (Supplier, Optional)'s internal value

```
AnnotatedType valueAnnotatedType = getOptionalValueAnnotatedType(property);

SingleElementProperty singleElementProperty = new SingleElementProperty(valueAnnotatedType);
```

and if only childProperty, is same behavior previous commit. work well

so i remove valueAnnotatedType parameter to SingleElementProperty constructor

> I'm not sure it's right time to split into OptionalProperty and SupplierProperty.

and I respect your comment, and I didn't separate it with confidence, only the feeling that role separation would be better, so i rollback to anonymous class

[8173376](https://github.com/naver/fixture-monkey/pull/951/commits/81733767bdc2bf552281f5c24b1d1f4b8d5faa42)